gordo_components/client/client.py

```diff
@@ -125,90 +120,92 @@ def __init__(
         """
 
-        original_endpoints = copy.copy(endpoints)
-
-        # Raise an error if we got some unhealty endpoints and weren't suppose to ignore them.
-        # otherwise filter out any unhealthy endpoints
-        if (
-            not target
-            and not ignore_unhealthy_targets
-            and not all(e.healthy for e in endpoints)
-        ):
-            raise ValueError(
-                f"Flag 'ignore_unhealthy_targets' is set to False and we encountered some "
-                f"unhealthy ones: {[ep for ep in endpoints if ep.healthy is False]}"
-            )
-        else:
-            endpoints = [ep for ep in endpoints if ep.healthy]
+        original_machines = copy.copy(machines)
 
-        # Filter down to single endpoint if requested
+        # Filter down to single machine if requested
         if target:
-            endpoints = [ep for ep in endpoints if ep.name == target]
+            machines = [ep for ep in machines if ep.name == target]
 
             # And check for single result and that it's healthy
-            if len(endpoints) != 1:
+            if len(machines) != 1:
                 raise ValueError(
-                    f"Found {'multiple' if len(endpoints) else 'no'} endpoints matching "
-                    f"target name '{target}' in {original_endpoints}"
-                )
-            if not endpoints[0].healthy:
-                raise ValueError(
-                    f"The targeted endpoint: '{endpoints[0]}' is unhealthy"
+                    f"Found {'multiple' if len(machines) else 'no'} machines matching "
+                    f"target name '{target}' in {original_machines}"
                 )
 
-        # finally, raise an error if all this left us without any endpoints
-        if not endpoints:
+        # finally, raise an error if all this left us without any machines
+        if not machines:
             raise ValueError(
-                f"Found no endpoints out of supplied endpoints: {original_endpoints} after filtering"
+                f"Found no machines out of supplied machines: {original_machines} after filtering"
             )
-        return endpoints
+        return machines
 
-    def _endpoints_from_watchman(self, endpoint: str) -> typing.List[EndpointMetadata]:
+    def _machines_from_server(self) -> typing.List[Machine]:
         """
-        Get a list of endpoints by querying Watchman
+        Get a list of machines by querying controller
         """
-        resp = self.session.get(endpoint)
+        resp = self.session.get(f"{self.server_endpoint}/models")
         if not resp.ok:
-            raise IOError(f"Failed to get endpoints: {repr(resp.content)}")
-        return [EndpointMetadata(data) for data in resp.json()["endpoints"]]
+            raise IOError(f"Failed to get machines: {repr(resp.content)}")
+        else:
+            model_names = resp.json()
+            with ThreadPoolExecutor(max_workers=self.parallelism) as executor:
```

Comment:
Probably makes sense to have this customizable as seperate parallelism, as fetching metadata is quite different from applying predictions.

Reply:
But we can do that later